File: 백준/Silver/1436. 영화감독 숌/영화감독 숌.cc
#include <iostream>
#include <vector>
#include <string>

using namespace std;

int main()
{
	vector<int> ans = { 666, 1666, 2666, 3666, 4666, 5666 };
	int n;

	cin >> n;
	if (n < 7) cout << ans[n-1];
	else {
		int answer = ans[5];
		while (n-6 > 0)
		{
			answer++;
			int cnt = 0;
			for (auto s : to_string(answer))
			{
				if (s == '6') cnt++;
				else cnt = 0;

				if (cnt == 3)
				{
					n--;
					break;
				}
			}
		}
		cout << answer;
	}

	return 0;
}

File: 백준/Silver/1436. 영화감독 숌/영화감독 숌.py
n = int(input())
ans = [666, 1666, 2666, 3666, 4666, 5666]

if n < 7:
    print(ans[n-1])
else:
    n = n - 6
    answer = ans[5]
    while (n > 0):
        # print("n : ", n)
        # print("answer : ", answer)
        answer += 1
        cnt = 0

        for i in str(answer):
            if i == '6':
                cnt += 1
            else:
                cnt = 0
            
            if cnt == 3:
                break
        
        if cnt == 3:
            n -= 1
        
    print(answer)